javascript/app-event-analysis/src/components/DimensionedLineChart.hooks.ts

```diff
@@ -0,0 +1,89 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import { useLayoutEffect, useRef, useState } from 'react'
+
+import type { PlaceboDataGroup, SynthControlData } from '../types.js'
+import type { ChartDimensions } from './ResultPane.types.js'
+
+const MIN_CONTAINER_HEIGHT = 800
+const SCROLL_BAR_OFFSET = 30
+
+export function useDynamicChartDimensions(
+	target: React.RefObject<HTMLDivElement> | null,
+	chartHeightPercent: number, // give the chart only a percentage of the total height
+	data: SynthControlData | Map<string, PlaceboDataGroup[]> | null,
+): ChartDimensions {
+	const widthDiff = useRef(0)
+	const heightDiff = useRef(0)
+	const innerOuterHeightDiff = useRef(0)
+
+	const [chartDimensions, setChartDimensions] = useState<ChartDimensions>({
+		width: 0,
+		height: 0,
+		margin: {
+			top: 1,
+			right: 100,
+			bottom: 50,
+			left: 100,
+		},
+	})
+
+	useLayoutEffect(() => {
```

Comment:
A ResizeObserver may make more sense. 

https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver

There are also hooks that utilize this (https://www.npmjs.com/package/use-resize-observer)